pkg/framework/didresolver/api.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package resolver
+
+import "time"
+
+// ResultType input option can be used to request a certain type of result.
+type ResultType int
+
+const (
+	// DidDocumentResult Request a DID Document as output
+	DidDocumentResult ResultType = iota
+	// ResolutionResult Request a DID Resolution Result
+	ResolutionResult
+)
+
+// didMethod operations
+type didMethod interface {
```

Comment:
@fqutishat how will users inject their own did methods if this interface is not exported?

Reply:
@llorllale func WithDidMethod(id string, method didMethod) Opt {

Reply:
the interface doesn't need to be exported for the exported function to accept it.

Reply:
@troyronda correct, but the user needs to know the shape of the interface in order to implement it

Reply:
I make it exported